tutorial.md

```diff
@@ -0,0 +1,171 @@
+machine.
+
+## Boson Tooling
+The primary interface for Boson project is the `faas` CLI.
+[Download][faas-download] the most recent version and install it some place
+within your `$PATH`.
+
+```sh
+# Be sure to download the correct binary for your operating system
+curl -L -o - faas.gz https://github.com/boson-project/faas/releases/download/v0.6.2/faas_linux_amd64.gz | gunzip > faas && chmod 755 faas
+sudo mv faas /usr/local/bin
+```
+## Configuring a Container Repository
+The unit of deployment in Boson Functions is an [OCI](https://opencontainers.org/)
+container image, typically referred to as a Docker container image.
+
+In order for the `faas` CLI to manage these containers, you'll need to be
+logged in to a container registry. For example, `docker.io/lanceball`
+
+
+```bash
+# Typically, this will log you in to docker hub if you
+# omit <repository.url>. If you are using a repository
+# other than Docker hub, provide that for <repository.url>
+docker login -u lanceball -p [redacted] <repository.url>
+```
+
+> Note: many of the `faas` CLI commands take a `--repository` argument.
+> Set the `FAAS_REPOSITORY` environment variable in order to omit this
+> parameter when using the CLI.
+
+```bash
+# This should be set to a repository that you have write permission
+# on and you have logged into in the previous step.
+export FAAS_REPOSITORY=docker.io/lanceball
+```
+
+## Creating a Project
+With your Knative enabled cluster up and running, you can now create a new
+Function Project. Let's start by creating a project directory. Function names
+in `faas` correspond to URLs at the moment, and there are some finicky cases
+at the moment. To ensure that everything works as it should, create a project
+directory consisting of three URL parts. Here is a good one.
+
+```bash
+mkdir fn.example.io
+cd fn.example.io
+```
+
+Now, with one command we will create the project files, build a container, and
+deploy the function as a Knative service.
+
+
+```bash
+faas create -l node -y
+```
+
+This will create a Node.js Function project in the current directory accepting
+all of the defaults inferred from your environment, for example`$FAAS_REPOSITORY`.
+When the command has completed, you can see the deployed function.
+
+```bash
+$ kn service list
+NAME            URL                                          LATEST                  AGE   CONDITIONS   READY   REASON
+fn-example-io   http://fn-example-io.faas.127.0.0.1.nip.io   fn-example-io-ngswh-1   24s   3 OK / 3     True
+```
+
+Clicking on the URL will take you to the running function in your cluster.
+
+## Local Development
+The `faas create` command also results in a docker container that can be run locally
+with container ports mapped to localhost.
+
+```bash
+faas run
+```
+
+For day to day development of the function, you can also run it locally outside of a
+container. For this project, using Node.js, you have the following commands available.
```

Comment:
should there be mention that you need node/npm installed?  I think this is the first time anything node.js related is shown, so it could be a dev from another language coming in.  Probably good to have it in the pre-req section?